src/Avalonia.Base/Input/PenDevice.cs

```diff
@@ -107,6 +108,18 @@ private void ProcessRawEvent(RawPointerEventArgs e)
             KeyModifiers inputModifiers, IInputElement? hitTest,
             Lazy<IReadOnlyList<RawPointerPoint>?>? intermediatePoints)
         {
+            if (pointer.CapturedGestureRecognizer is GestureRecognizer gestureRecognizer)
+            {
+                if (gestureRecognizer.Target != null)
+                {
+                    var e = new PointerEventArgs(InputElement.PointerMovedEvent, gestureRecognizer.Target, pointer, (Visual)root,
```

Comment:
Why can't we use `gestureRecognizer.Target` as the source below? e. g.
`var source = gestureRecognizer.Target ?? pointer.Captured ?? hitTest)`.
And then add a check before `source.RaiseEvent`? That would reduce the amount of copypasta